engine/src/flutter/lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -1018,6 +1032,158 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
+  /// recalculate [lineHeightScaleFactorOverride], [letterSpacingOverride],
+  /// [wordSpacingOverride], and [paragraphSpacing].
+  ///
+  /// Updates [lineHeightScaleFactorOverride], [letterSpacingOverride],
+  /// [wordSpacingOverride], and [paragraphSpacing] with the new values.
+  DomResizeObserver? _typographySettingsObserver;
+  DomElement? _typographyMeasurementElement;
+
+  /// Updates [lineHeightScaleFactorOverride] and return true if
+  /// [lineHeightScaleFactorOverride] changed. If not then returns false.
+  bool _updateLineHeightScaleFactorOverride(double? value) {
+    if (configuration.lineHeightScaleFactorOverride != value) {
+      configuration = configuration.apply(lineHeightScaleFactorOverride: value);
+      return true;
+    }
+    return false;
+  }
+
+  /// Updates [letterSpacingOverride] and return true if
+  /// [letterSpacingOverride] changed. If not then returns false.
+  bool _updateLetterSpacingOverride(double? value) {
+    if (configuration.letterSpacingOverride != value) {
+      configuration = configuration.apply(letterSpacingOverride: value);
+      return true;
+    }
+    return false;
+  }
+
+  /// Updates [wordSpacingOverride] and returns true if
+  /// [wordSpacingOverride] changed. If not then returns false.
+  bool _updateWordSpacingOverride(double? value) {
+    if (configuration.wordSpacingOverride != value) {
+      configuration = configuration.apply(wordSpacingOverride: value);
+      return true;
+    }
+    return false;
+  }
+
+  /// Updates [paragraphSpacing] and returns true if
+  /// [paragraphSpacing] changed. If not then returns false.
+  bool _updateParagraphSpacing(double value) {
+    if (configuration.paragraphSpacing != value) {
+      configuration = configuration.apply(paragraphSpacing: value);
+      return true;
+    }
+    return false;
+  }
+
+  /// Set the callback function for updating [lineHeightScaleFactorOverride],
+  /// [letterSpacingOverride], [wordSpacingOverride], and [paragraphSpacing]
+  /// based on the sizing changes of an off-screen element with text.
+  void _addTypographySettingsObserver() {
+    _typographyMeasurementElement = createDomHTMLParagraphElement();
+    _typographyMeasurementElement!.text = 'flutter typography measurement';
+    // The element should be hidden from screen readers.
+    _typographyMeasurementElement!.setAttribute('aria-hidden', 'true');
+    final DomCSSStyleDeclaration style = _typographyMeasurementElement!.style;
+    style.position = 'absolute';
+    // The element should be off-screen and not visible.
+    style.top = '-9999px';
+    style.left = '-9999px';
+    style.visibility = 'hidden';
+    // The element should be sensitive to letter-spacing, word-spacing,
+    // and line-height changes.
+    style.width = 'auto';
+    style.height = 'auto';
+    style.whiteSpace = 'nowrap';
+    // Set text spacing properties defaults.
+    const double spacingDefault = 100.0;
```

Comment:
Would it make sense to use 9999 here as well? 100px seems like a plausible real value a user might set for accessibility.

Reply:
Generally speaking, a very high number like 9999 should work. But in this case, the element is positioned at `-9999px` too, which means the element could become tall enough to reach the top edge of the window and interfere with the page.

Visually, it's hidden to it won't interfere with rendering. But it will interfere with pointer events. I suggest adding:
```dart
..pointerEvents = 'none';
```
above to make the element transparent to pointer events too.